stdlib/dynarray.mli

```diff
@@ -0,0 +1,190 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                            Simon Cruanes                               *)
+(*                                                                        *)
+(*   Copyright 2022 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Growable, mutable array.
+
+
+
+    @since 5.1
+*)
+
+type 'a t
+(** A dynamic array containing values of type ['a].
+
+    This contains an underlying {!array} along with a size.
+    Operations such as {!push}, {!append}, and {!append_seq}, extend the
+    size (and might reallocate the underlying array).
```

Comment:
I'm not interested in how you implement this. I think it's better to indicate the general space and time properties of the datastructure.

Reply:
I am very interested in the underlying behavior of the implementation, and that `get` is basically similar to a `Array.get`. 

Other data structures do the same:
- `Map`: 
  > The implementation uses balanced binary trees, and therefore searching
  > and insertion take time logarithmic in the size of the map.
- `Hashtbl.create` details how the table grows underneath as needed, and provides `reset`/`clear` which differ only by how they affect the underlying array.
- `Buffer` specifies
  > This module implements buffers that automatically expand as necessary. 

Moreover, knowing the allocation profile of such a core data structure is important. People will [depend](https://xkcd.com/1172/) on it even if it's not specified. Someone recently told me:

> I often see programmers getting terrible performance out of their programs because they create little allocation hells for themselves.

Reply:
> Moreover, knowing the allocation profile of such a core data structure is important.

Indeed that's what I meant by space properties.